### PR TITLE
fix(autoware_obstacle_cruise_planner): fix knownConditionTrueFalse warnings

### PR DESCRIPTION
**Pull request by veqcc**
## Description

This is a fix based on cppcheck `knownConditionTrueFalse` warning

```
planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp:351:18: style: Condition 'stop_dist' is always true [knownConditionTrueFalse]
      } else if (stop_dist) {
                 ^
```

and 

```
planning/autoware_obstacle_cruise_planner/src/planner_interface.cpp:640:49: style: Condition 'slow_down_end_idx' is always true [knownConditionTrueFalse]
    slow_down_debug_multi_array_.data.push_back(slow_down_end_idx ? *slow_down_end_idx : -1.0);
                                                ^
```

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp [hunk 1]
```diff
@@ -348,7 +348,7 @@ std::tuple<double, double> OptimizationBasedPlanner::calcInitialMotion(
           "engage speed (%.3f) until vehicle speed reaches engage_vel_thr (%.3f). stop_dist = %.3f",
           vehicle_speed, target_vel, engage_velocity_, engage_vel_thr, stop_dist.value());
         return std::make_tuple(initial_vel, initial_acc);
-      } else if (stop_dist) {
+      } else {
```
Comment (kosuke55) [inline planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp:351]: the type of stop_dist is std::optional<double>.
I am wondering if this warning is correct?

Comment (kosuke55) [reply planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp:351]: or should we use `stop_dist.has_value()` ?

Comment (veqcc) [reply planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp:351]: The original if branch is like this:
```
if ( ... && !stop_dist) {
  ...
} else if (stop_dist) {
  ...
}
```
if ` ... && !stop_dist` if false, then I think `stop_dist` is true.
Isn't is correct?

Comment (kosuke55) [reply planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp:351]: ah, you are right :bow:

Comment (veqcc) [reply planning/autoware_obstacle_cruise_planner/src/optimization_based_planner/optimization_based_planner.cpp:351]: > or should we use stop_dist.has_value() ?

Yes if you are not sure how `!stop_dist` is evaluated.

**Review by kosuke55 — APPROVED** [review]

**Comment by veqcc** [comment]
@kosuke55 
Btw, the if true branch is like this:
```
    if ((stop_dist && *stop_dist > stop_dist_to_prohibit_engage_) || !stop_dist) {
        initial_vel = engage_velocity_;
        initial_acc = engage_acceleration_;
        RCLCPP_DEBUG(
          rclcpp::get_logger("ObstacleCruisePlanner::OptimizationBasedPlanner"),
          "calcInitialMotion : vehicle speed is low (%.3f), and desired speed is high (%.3f). Use "
          "engage speed (%.3f) until vehicle speed reaches engage_vel_thr (%.3f). stop_dist = %.3f",
          vehicle_speed, target_vel, engage_velocity_, engage_vel_thr, stop_dist.value());
        return std::make_tuple(initial_vel, initial_acc);
      }
```

When `!stop_dist` is true, then what will happen for `stop_dist.value()` ?

**Comment by kosuke55** [comment]
@veqcc 
> When !stop_dist is true, then what will happen for stop_dist.value() ?

oh, thanks, this may cause crash.  May I ask you to fix it too?

**Comment by veqcc** [comment]
@kosuke55 
Sure!! Could you tell me the expected behavior?
(e.g. just eliminate from DEBUG print)

**Comment by kosuke55** [comment]
@veqcc 
We would like to keep the current behavior as much as possible like branching whether to show only stop_dist or not.
The following example is dirty but I have heard that stringstream is heavy, so if you have any better to way please apply that.

```
      if ((stop_dist && *stop_dist > stop_dist_to_prohibit_engage_) || !stop_dist) {
        initial_vel = engage_velocity_;
        initial_acc = engage_acceleration_;
        std::string message =
          "calcInitialMotion : vehicle speed is low (%.3f), and desired speed is high (%.3f). Use "
          "engage speed (%.3f) until vehicle speed reaches engage_vel_thr (%.3f)";
        if (stop_dist) {
          message += ". stop_dist = %.3f";
          RCLCPP_DEBUG(
            rclcpp::get_logger("ObstacleCruisePlanner::OptimizationBasedPlanner"), message.c_str(),
            vehicle_speed, target_vel, engage_velocity_, engage_vel_thr, stop_dist.value());
        } else {
          RCLCPP_DEBUG(
            rclcpp::get_logger("ObstacleCruisePlanner::OptimizationBasedPlanner"), message.c_str(),
            vehicle_speed, target_vel, engage_velocity_, engage_vel_thr);
        }
        return std::make_tuple(initial_vel, initial_acc);
```

**Pull request merged**